primitives/weights/src/weight_v2.rs

```diff
@@ -230,14 +300,20 @@ impl Zero for Weight {
 impl Add for Weight {
```

Comment:
Not sure if we could make this `std` only, since the runtime should not use unsafe math anyway.

Reply:
Not really unsafe math.